scanner/src/main/kotlin/LocalScanner.kt

```diff
@@ -136,17 +136,12 @@ abstract class LocalScanner(name: String, config: ScannerConfiguration) : Scanne
      */
     fun getDetails() = ScannerDetails(scannerName, getVersion(), getConfiguration())
 
-    // `newFixedThreadPoolContext` is obsolete and will be replaced with a new API. Keep using it until the new API is
-    // available, see:
-    // https://kotlin.github.io/kotlinx.coroutines/kotlinx-coroutines-core/kotlinx.coroutines/new-fixed-thread-pool-context.html
-    // https://github.com/Kotlin/kotlinx.coroutines/issues/261
-    @ObsoleteCoroutinesApi
     override suspend fun scanPackages(packages: List<Package>, outputDirectory: File, downloadDirectory: File):
             Map<Package, List<ScanResult>> {
         val scannerDetails = getDetails()
 
-        val storageDispatcher = newFixedThreadPoolContext(5, "storage")
-        val scanDispatcher = newFixedThreadPoolContext(1, "scan")
+        val storageDispatcher = Executors.newFixedThreadPool(5).asCoroutineDispatcher()
+        val scanDispatcher = Executors.newSingleThreadExecutor().asCoroutineDispatcher()
```

Comment:
Hmm, it's a pity that we don't seem to have a way anymore to express a name for thread pool / executor :cry:

Reply:
There is an option by providing a custom thread factory. I'll check this and make another PR if it works.